### DISCUSSION TITLE
Question on culling

### DISCUSSION THREAD

**Discussion by zwalden** [spimort/TerraBrush#103] [General]
Hello! Godot newbie, so pardon if this is clear already; but I understand that Godot4 handles frustum culling for us, so things not in viewport are not rendered, from what I understand, my two questions are:

# frustum culling still occurs here, right?
# what about occlusion culling? 

Very cool plugin!

**Comment by zwalden** [comment]
Small follow up, how does collision work for the landscape/placed scattered items?

**Comment by spimort** [comment]
Hey @zwalden ! I'm glad you like the plugin!

For the frustum culling, as far as I know, it still occurs, yeah!
For the occlusion culling, I've not mess a lot with it, but I think it requires the objects to be direct child of the baker?

Tho, using the occlusion culling has cost on the CPU so I don't know if this is the best option.

What I can suggest is to use the visibility range on the mesh when you create a packed scene with your mesh : 
![image](https://github.com/spimort/TerraBrush/assets/6035485/f4bb9179-42be-4d22-8eeb-2ad020b33ba9)

Also, for the collision, this is the same thing, once you created a packed scene with your mesh, you can add collision to it. Make sure that the object you drag in the object section is the packed scene and not the mesh.

I hope this helps!
